src/test/java/io/knotx/launcher/KnotxStarterVerticleStoresTest.java

```diff
@@ -0,0 +1,178 @@
+        .subscribe(
+            success -> testContext.completeNow(),
+            testContext::failNow
+        );
+  }
+
+  @Test
+  @DisplayName("Expect instance running, when starting with one not-optional and present store")
+  void startWithOneMandatoryPresentStore(VertxTestContext testContext, Vertx vertx) {
+    // given
+    DeploymentOptions options = fromBootstrapFile(
+        "multiple-stores/bootstrap/one-present-mandatory-store.json");
+
+    vertx.registerVerticleFactory(TestVerticlesFactory
+        .allVerticlesStart(
+            VerificationContext.instance()
+                // then
+                .setAssertions(validateTestOptionValue("testValue"))
+                .setTestContext(testContext))
+    );
+
+    // when
+    vertx.rxDeployVerticle(KnotxStarterVerticle.class.getName(), options)
+        .subscribe(
+            success -> testContext.completeNow(),
+            testContext::failNow
+        );
+  }
+
+  @Test
+  @DisplayName("Expect instance failing, when starting with one non-optional and missing store")
+  void startWithOneMandatoryMissingStore(VertxTestContext testContext, Vertx vertx) {
+    // given
+    DeploymentOptions options = fromBootstrapFile(
+        "multiple-stores/bootstrap/one-missing-mandatory-store.json");
+
+    // when
+    vertx.rxDeployVerticle(KnotxStarterVerticle.class.getName(), options)
+        .subscribe(
+            s -> testContext.failNow(new RuntimeException("This deployment should fail")),
+            throwable -> {
+              assertTrue(throwable instanceof FileSystemException);
+              assertTrue(throwable.getMessage().contains("file-that-does-not-exists.conf"));
+              testContext.completeNow();
+            }
+        );
+  }
+
+  @Test
+  @DisplayName("Expect configuration overridden by second store")
+  void checkConfigurationOverriding(VertxTestContext testContext, Vertx vertx) {
+    // given
+    DeploymentOptions options = fromBootstrapFile(
+        "multiple-stores/bootstrap/two-stores-with-overrides.json");
+
+    vertx.registerVerticleFactory(TestVerticlesFactory
+        .allVerticlesStart(
+            VerificationContext.instance()
+                // then
+                .setAssertions(jsonObject -> {
+                  assertEquals("testValue-overridden", jsonObject.getString(CONFIG_TEST_OPTION_KEY));
+                  assertEquals("knotx", jsonObject.getString("defaultOption"));
+                  assertEquals("rocks", jsonObject.getString("extraOption"));
+                })
+                .setTestContext(testContext))
+    );
+
+    // when
+    vertx.rxDeployVerticle(KnotxStarterVerticle.class.getName(), options)
+        .subscribe(
+            success -> testContext.completeNow(),
+            testContext::failNow
+        );
+  }
+
+  @Test
+  @DisplayName("Expect all modules deployed from several stores")
+  void checkAllModulesDeployed(VertxTestContext testContext, Vertx vertx) {
```

Comment:
This is still not working. Additionally test requires more assertions because it is green but modules were not deployed.